app/src/main/java/com/example/learn09082021v001/ReadAndWriteMemori.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.learn09082021v001
+
+import android.content.Context
+import android.widget.Toast
+import java.lang.Exception
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.BufferedReader
+import java.io.FileInputStream
+import java.io.InputStreamReader
+
+class ReadAndWriteMemori {
+    var dataMemoriLine1 = ""
+    var dataMemoriLine2 = ""
+    var dataMemoriLine3 = ""
+    var dataMemoriLine4 = ""
+    var dataMemoriLine5 = ""
+    var dataMemoriLine6 = ""
+    var dataMemoriLine7 = ""
+
+    fun WriteDataMemory(
+        context: Context,
+        fileNameSaverMemory: String,
+        dataSaverMemori: String = "",
+    ) {
+        val fileOutputStream: FileOutputStream
+        try {
+            fileOutputStream = context.openFileOutput(fileNameSaverMemory, Context.MODE_PRIVATE)
+            fileOutputStream.write(dataSaverMemori.toByteArray())
+            Toast.makeText(context, "data update complete", Toast.LENGTH_SHORT).show()
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
+            println("data wite fail")
+        } catch (e: Exception) {
+            e.printStackTrace()
+            println("data wite fail")
```

Comment:
after use any stream, you have to call `fileOutputSteam` to close, to avoid memory leak
Kotlin already support a short way to do it, you can check ` try-with-resources `